crates/nu-command/src/system/run_external.rs

```diff
@@ -559,12 +559,11 @@ impl ExternalCommand {
                                         "{cause}: child process '{commandname}' core dumped"
                                     );
                                     eprintln!("{}", style.paint(&message));
-                                    let _ = exit_code_tx.send(Value::error(
-                                        ShellError::ExternalCommand {
-                                            label: "core dumped".into(),
-                                            help: message,
-                                            span: head,
-                                        },
+                                    let _ = exit_code_tx.send(Value::int(
+                                        // If we don't get a code (unlikely)
```

Comment:
This is highly likely. On unix, if a process is killed via signal, then [`code()` will be `None`](https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.code). 

It's probably out of scope for this PR, but this should match what we do for signals when the process doesn't core dump (right now, we always return -1).